dali/operators/geometry/affine_transforms/transform_crop.cc

```diff
@@ -0,0 +1,153 @@
+    DALI_FLOAT_VEC, true)
+  .AddArg(
+    "to_end",
+    R"code(The upper bound of the destination coordinate space.)code",
+    DALI_FLOAT_VEC, true)
+  .AddOptionalArg(
+    "absolute",
+    R"code(If set to true, start and end coordinates will be swapped if start > end.)code",
+    false, true)
+  .NumInput(0, 1)
+  .NumOutput(1)
+  .AddParent("TransformAttr");
+
+/**
+ * @brief Scale transformation.
+ */
+class TransformCropCPU
+    : public TransformBaseOp<CPUBackend, TransformCropCPU> {
+ public:
+  using SupportedDims = dims<1, 2, 3, 4, 5, 6>;
+
+  explicit TransformCropCPU(const OpSpec &spec) :
+      TransformBaseOp<CPUBackend, TransformCropCPU>(spec),
+      from_start_("from_start", spec),
+      from_end_("from_end", spec),
+      to_start_("to_start", spec),
+      to_end_("to_end", spec) {
+  }
+
+  template <typename T, int mat_dim>
+  void DefineTransforms(span<affine_mat_t<T, mat_dim>> matrices) {
+    constexpr int ndim = mat_dim - 1;
+    assert(matrices.size() == static_cast<int>(from_start_.size()));
+    assert(matrices.size() == static_cast<int>(from_end_.size()));
+    assert(matrices.size() == static_cast<int>(to_start_.size()));
+    assert(matrices.size() == static_cast<int>(to_end_.size()));
+    for (int i = 0; i < matrices.size(); i++) {
+      auto &mat = matrices[i];
+      const vec<ndim> &from_start = *reinterpret_cast<vec<ndim>*>(from_start_[i].data());
+      const vec<ndim> &from_end = *reinterpret_cast<vec<ndim>*>(from_end_[i].data());
+      const vec<ndim> &to_start = *reinterpret_cast<vec<ndim>*>(to_start_[i].data());
+      const vec<ndim> &to_end = *reinterpret_cast<vec<ndim>*>(to_end_[i].data());
+      mat = affine_mat_t<T, mat_dim>::identity();
+      vec<ndim> from_extent = from_end - from_start;
+      vec<ndim> to_extent = to_end - to_start;
+      vec<ndim> offset = to_start - from_start;
+
+      if (absolute_) {
+        for (int d = 0; d < ndim; d++) {
+          from_extent[d] = std::abs(from_extent[d]);
+          to_extent[d] = std::abs(to_extent[d]);
+        }
+      }
+
+      for (int d = 0; d < ndim; d++) {
+        constexpr float kEps = std::numeric_limits<float>::epsilon();
+        DALI_ENFORCE(std::abs(from_extent[d]) > kEps,
+          make_string("from_end[d] should be different from from_start[d] for all "
+                      "dimensions. Got: from_start=",
+                      from_start, " and from_end=", from_end,
+                      ". Note: the two numbers are considered equal if their absolute "
+                      "difference is equal or smaller than ", kEps));
+        DALI_ENFORCE(std::abs(to_extent[d]) > kEps,
+          make_string("to_end[d] should be different from to_start[d] for all "
+                      "dimensions. Got: to_start=",
+                      to_start, " and to_end=", to_end,
+                      ". Note: the two numbers are considered equal if their absolute "
+                      "difference is equal or smaller than ", kEps));
```

Comment:
It's not strictly necessary - the matrix will be singular, but the operation _can_ be performed and makes some sense.

Reply:
OK